buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/transform/RestTestTransform.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.test.rest.transform;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * A single instruction to transforms a REST test.
+ */
+public interface RestTestTransform<T extends JsonNode> {
```

Comment:
Is there a reason we can't merge this interface with `RestTestTransformGlobalSetup` and `RestTestTransformGlobalTeardown`. Implementations need not actually implement those methods as we could provide default noop implementations in the interface itself or in an abstract base class.

Doing so would eliminate all the nasty `instanceof` runtime checks in `RestTestTransformer`.